### PR TITLE
Add godocs, comments and README contents

### PR DESCRIPTION
**Pull request by MDr164**
Signed-off-by: Marvin Drees <marvin.drees@9elements.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+
+## Requirements
+
+Go version 1.17 or higher.
+
+---
+
+## Installtion instructions
+
+Either run `go install github.com/system/transparency` or clone the repo and run `go build`.
```
Comment (sylv-io) [inline README.md]: `go install github.com/system-transparency/stmgr@latest`

**Review by sylv-io — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+
+---
+
+## Requirements
+
+Go version 1.17 or higher.
+
+---
+
+## Installtion instructions
```
Comment (sylv-io) [inline README.md]: Installation

File: README.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+
+Either run `go install github.com/system-transparency/stmgr@latest` or clone the repo and run `go build`.
+
+---
+
+## Usage
+
+This tool requires to be invoked with a command and corresponding subcommand.
```
Comment (sylv-io) [inline README.md:25]: invocation

File: README.md [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+That means an example invokation looks like this: `stmgr ospkg create`.
+In that example `ospkg` is the command and `create` is the subcommand.
```
Comment (sylv-io) [inline README.md]: In that example, `ospkg` ...

File: eval/keygen.go [hunk 5]
```diff
@@ -7,8 +7,12 @@ import (
 	"github.com/system-transparency/stmgr/log"
 )
 
+// KeygenCertificate takes arguments like os.Args as a string array
+// and maps them to their corresponding flags using the std flag
+// package. It the calls keygen.Certificate after they are parsed.
```
Comment (sylv-io) [inline eval/keygen.go]: It then calls keygen. Certificate after they are parsed.
-------^------------------^

File: eval/ospkg.go [hunk 6]
```diff
@@ -7,7 +7,11 @@ import (
 	"github.com/system-transparency/stmgr/ospkg"
 )
 
+// OspkgCreate takes arguments like os.Args as a string array
+// and maps them to their corresponding flags using the std flag
+// package. It the calls ospkg.Create after they are parsed.
```
Comment (sylv-io) [inline eval/ospkg.go]: It then calls ospkg. Create after they are parsed.
-------^----------------^

File: eval/ospkg.go [hunk 7]
```diff
@@ -44,28 +52,30 @@ func OspkgCreate(args []string) error {
 	)
 }
 
+// OspkgSign takes arguments like os.Args as a string array
+// and maps them to their corresponding flags using the std flag
+// package. It the calls ospkg.Sign after they are parsed.
```
Comment (sylv-io) [inline eval/ospkg.go]: same here

File: eval/provision.go [hunk 8]
```diff
@@ -8,7 +8,11 @@ import (
 	"github.com/system-transparency/stmgr/provision"
 )
 
+// ProvisionHostconfig takes arguments like os.Args as a string array
+// and maps them to their corresponding flags using the std flag
+// package. It the calls provision.Cfgtool after they are parsed.
```
Comment (sylv-io) [inline eval/provision.go]: same here

File: keygen/certificate.go [hunk 9]
```diff
@@ -69,6 +78,7 @@ func Certificate(args *CertificateArgs) error {
 	)
 
 	if len(args.RootCertPath) == 0 {
+		// Create a selfsigned certificate.
```
Comment (sylv-io) [inline keygen/certificate.go]: self-signed

File: keygen/certificate.go [hunk 10]
```diff
@@ -106,6 +117,7 @@ func checkArgs(args *CertificateArgs) error {
 	}
 }
 
+// This function makes sure the ondisk format of the key and certificate are correct.
```
Comment (sylv-io) [inline keygen/certificate.go]: on-disk

File: keygen/certificate.go [hunk 11]
```diff
@@ -223,6 +235,9 @@ func parseValidUntil(date string) (time.Time, error) {
 	return time.Parse(time.RFC822, date)
 }
 
+// This is the core function to create certificate and the corresponding ed25519 key.
```
Comment (sylv-io) [inline keygen/certificate.go]: a certificate ?
not sure

File: keygen/certificate.go [hunk 12]
```diff
@@ -223,6 +235,9 @@ func parseValidUntil(date string) (time.Time, error) {
...
+// It uses golangs std crypto package and uses a cryptographcially secure (enough) source
```
Comment (sylv-io) [inline keygen/certificate.go]: Golangs

Comment (sylv-io) [reply keygen/certificate.go]: cryptographically

Comment (sylv-io) [reply keygen/certificate.go]: of entropy, namely

File: keygen/util.go [hunk 13]
```diff
@@ -14,6 +14,8 @@ var (
 
 const defaultFilePerm fs.FileMode = 0o600
 
+// LoadPEM reads a PEM formatted file from disk
+// and returns a pointer of the pem.Block data.
```
Comment (sylv-io) [inline keygen/util.go]: a pointer to the ..

File: ospkg/create.go [hunk 14]
```diff
@@ -18,6 +18,10 @@ type CreateArgs struct {
 	Cmdline   string
 }
 
+// Create will create a new OS package using the
+// stboot/ospkg package. If no errors occur it will
```
Comment (sylv-io) [inline ospkg/create.go]: occur, it

File: provision/cfgtool.go [hunk 15]
```diff
@@ -5,6 +5,14 @@ import (
 	"time"
 )
 
+// Cfgtool is used to provision a stboot node by creating a
+// host configuration json file. If any value of the host
```
Comment (sylv-io) [inline provision/cfgtool.go]: JSON

File: provision/parser.go [hunk 16]
```diff
@@ -11,6 +11,11 @@ import (
 
 const defaultFilePerm fs.FileMode = 0o600
 
+// HostCfgSimplified mirrors what stboots host configuration
+// looks like but does not implement their custom types and
```
Comment (sylv-io) [inline provision/parser.go]: , but

Comment (sylv-io) [reply provision/parser.go]: , but

File: provision/tui.go [hunk 17]
```diff
@@ -43,6 +43,10 @@ var (
 	extension = make(map[string]string) //nolint:gochecknoglobals
 )
 
+// This function constructs the and manages the terminal UI application.
+// It uses tview for that as it abstracts tcell functions and makes building
```
Comment (sylv-io) [inline provision/tui.go]: that, as

File: provision/tui.go [hunk 18]
```diff
@@ -43,6 +43,10 @@ var (
...
+// simple UIs easier. The downside is, that this UI is stateful and needs
+// a few linter breaking code idioms which you can find in as nolint stanzas.
```
Comment (sylv-io) [inline provision/tui.go]: idioms, which

**Review by sylv-io — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (MDr164) [reply README.md]: Ack

File: README.md (hunk 3 above)
Comment (MDr164) [reply README.md:25]: I don't think this word makes any sense in this sentence. Mind telling me where you wanted to put it?

File: README.md (hunk 4 above)
Comment (MDr164) [reply README.md]: Ack

File: eval/keygen.go (hunk 5 above)
Comment (MDr164) [reply eval/keygen.go]: There is deliberately no space after the dot, as this is a function name and not a sentence end. The missing n is indeed missing though.

File: eval/ospkg.go (hunk 6 above)
Comment (MDr164) [reply eval/ospkg.go]: See above

File: eval/ospkg.go (hunk 7 above)
Comment (MDr164) [reply eval/ospkg.go]: See above

File: eval/provision.go (hunk 8 above)
Comment (MDr164) [reply eval/provision.go]: See above

File: keygen/certificate.go (hunk 9 above)
Comment (MDr164) [reply keygen/certificate.go]: Ack

File: keygen/certificate.go (hunk 10 above)
Comment (MDr164) [reply keygen/certificate.go]: Ack

File: keygen/certificate.go (hunk 11 above)
Comment (MDr164) [reply keygen/certificate.go]: Ack

File: keygen/certificate.go (hunk 12 above)
Comment (MDr164) [reply keygen/certificate.go]: Ack

File: keygen/util.go (hunk 13 above)
Comment (MDr164) [reply keygen/util.go]: Ack

File: ospkg/create.go (hunk 14 above)
Comment (MDr164) [reply ospkg/create.go]: Ack

File: provision/cfgtool.go (hunk 15 above)
Comment (MDr164) [reply provision/cfgtool.go]: Ack

File: README.md (hunk 3 above)
Comment (sylv-io) [reply README.md:25]: Indeed, sorry :D

File: provision/parser.go (hunk 16 above)
Comment (MDr164) [reply provision/parser.go]: Ack

File: provision/tui.go (hunk 18 above)
Comment (MDr164) [reply provision/tui.go]: Ack

File: provision/tui.go (hunk 17 above)
Comment (MDr164) [reply provision/tui.go]: Ack

**Review by sylv-io — APPROVED** [review]

**Pull request merged**